Modules/hpe3par_snapshot.py

```diff
@@ -435,13 +468,188 @@ def restore_snapshot_online(
+        return (False, False, "Schedule create failed. Base volume name must be atleast 1 character and not more than 31 characters", {})
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        expirationHours = convert_to_hours(expiration_time, expiration_unit)
+        retentionHours = convert_to_hours(retention_time, retention_unit)
+
+        freq = "@hourly"
+        if not client_obj.volumeExists(base_volume_name):
+           return (False, False, "Volume does not Exist", {})
+        
+        if retentionHours > expirationHours:
+           return (False, False, "Expiration time must be greater than or equal to retention time", {})   
+       
+        if task_freq_custom:          
+          if ' ' in task_freq_custom:            
+           taskcm = str(task_freq_custom).split()
+           if len(taskcm) == 5:
+               minutes_t = taskcm[0]               
+               hour_t = taskcm[1]
+               day_t = taskcm[2]
+               month_t = taskcm[3]  
+               dow_t = taskcm[4]
+               if '*' not in minutes_t:
+                  if (int(minutes_t) > 50 or int(minutes_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(minutes_t)>1 or minutes_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in hour_t:
+                  if (int(hour_t) > 23 or int(hour_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(hour_t)>1 or hour_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in day_t:
+                  if (int(day_t) > 31 or int(day_t) < 1):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(day_t)>1 or day_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in month_t:
+                  if (int(month_t) > 12 or int(month_t) < 1):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(month_t)>1 or month_t == ""):
+                    return (False, False, "Invalid task freq", {})
+               if '*' not in dow_t:
+                  if (int(dow_t) > 6 or int(dow_t) < 0):
+                    return (False, False, "Invalid task freq", {})
+               else:
+                  if (len(dow_t)>1 or dow_t == ""):
+                    return (False, False, "Invalid task freq", {})
+           else:
+              return (False, False, "Invalid task freq", {})
+          else:
+              return (False, False, "Invalid task freq", {})
+          freq = task_freq_custom
+
+        if task_freq_custom and task_freq:
```

Comment:
Why is this check after we login? Shouldn't it be before we login along with the rest of the pre checks